### PR TITLE
Fix scope definition in ES 100.

### PR DESCRIPTION
**Pull request by ShchchowAMD**
1. Add tokens related to IMAGE SHADOW.

2. Allow redefinition with a same name in parameter list. Only in ES 100.
" If it is declared as a parameter in a function definition, it is scoped until the end of that function definition.  A function body has a scope nested inside the function’s definition."

### CODE REVIEW

**Comment by johnkslang** [comment]
But, these types don't exist.  It will be backward incompatible.  What's missing here?

**Comment by ShchchowAMD** [comment]
I‘ll do further check.
Seems they should be all reserved.

**Comment by ShchchowAMD** [comment]
Hi John,

I found image2DShadow and other 3 tokens were in the reserved key words list from version 140 to 410.

If I understand right, those four removed tokens should not be added back to grammar right?

**Comment by ShchchowAMD** [comment]
Remove those 4 tokens.

Test case for scope definition in Version 100 is : 

```
 #version 100 es
 int func (int inp, int x)
  { 
 
 int x = 5; return inp + x - 5; 
 
 }
``` 

should be treated as 

```
#version 100 es
int func (int inp, int x) 
{ 
     {
           int x = 5; return inp + x - 5; 
     }
}
```

**Comment by johnkslang** [comment]
Can I ask what's motivating changing the way scoping works?  Is it only from reading the specification, or is it something else as well?

**Comment by ShchchowAMD** [comment]
Hi John,

Those issues are found during debugging with CTS test cases. The motivation is to pass the test if something against the specification.

Related part of cases' shader sources are attached in previous comment.

**Comment by ShchchowAMD** [comment]
Above shaders could be found in CTS cases:
dEQP-GLES2.functional.shaders.scoping.valid.local_variable_hides_function_parameter_vertex
dEQP-GLES2.functional.shaders.scoping.valid.local_variable_hides_function_parameter_fragment

File: Test/baseResults/100scope.vert.out [hunk 1]
```diff
@@ -1,5 +1,4 @@
 100scope.vert
-ERROR: 0:5: 'a' : redefinition
```
Comment (johnkslang) [inline Test/baseResults/100scope.vert.out:2]: Do we still have other tests that are testing this for the normal case?  I don't want to lose this test.  Is ES 100 the only exception to the normal scoping rules?

**Comment by ShchchowAMD** [comment]
If I understand right, this only occurs when version = 100 ES, same as declared version in cts case.

Maybe I can copy this point to version 300 and add comments, which could also show the difference.

**Comment by johnkslang** [comment]
Yes, or somehow verify we test the other scoping rule.

**Comment by ShchchowAMD** [comment]
Hi john, we could found same codes in 110scope.vert.

They could be well compared to see the difference of redefinition scoping in both versions.

**Pull request merged**